### PR TITLE
CD, CI, Test Coverage document

### PR DESCRIPTION
**Pull request by oliverochman**
[PT link](https://www.pivotaltracker.com/story/show/169889114)
PT description:
```
Finish writing the guide, images and structure are already there.

This is a NEW guide that is supposed to be added to Week 4.
```

### CODE REVIEW

File: guides/coveralls-ci-cd/coveralls_ci_cd.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
-**The following guide assumes you're using both RSpec and Cucumber for testing.**
+ **The following guide assumes you're using both RSpec and Cucumber for testing.**
+
+**The following guide assumes that you have an upstream repository on Craft Academys Github account, if that is not the case for you then just ignore the parts about adding the "Craft Academy repository" and set your own repository there instead**
 
 We'll make use of [Coveralls](https://coveralls.io/) a web service that helps track code coverage.
-You can use it for various programming languages and for Ruby, there is a gem we can use.
+You can use it for various programming languages and Ruby, there is a gem we can use.
```
Comment (GergKllai1) [inline guides/coveralls-ci-cd/coveralls_ci_cd.md]: wording

File: guides/coveralls-ci-cd/coveralls_ci_cd.md [hunk 2]
```diff
@@ -1,7 +1,9 @@
-**The following guide assumes you're using both RSpec and Cucumber for testing.**
+ **The following guide assumes you're using both RSpec and Cucumber for testing.**
+
+**The following guide assumes that you have an upstream repository on Craft Academys Github account, if that is not the case for you then just ignore the parts about adding the "Craft Academy repository" and set your own repository there instead**
```
Comment (GergKllai1) [inline guides/coveralls-ci-cd/coveralls_ci_cd.md]: repetition of  "The following guide assumes"

File: guides/coveralls-ci-cd/coveralls_ci_cd.md [hunk 3]
```diff
@@ -44,8 +46,7 @@ we should use
 Coveralls.wear_merged!('rails')
 ```
 
-Then create a custom rake task that will run all your test suites then submit coverage
-results to the coveralls website.
+Then create a custom rake task that will run all your test suites then submit coverage results to coveralls.
 
 Create a new file `lib/tasks/ci.rake`
```
Comment (GergKllai1) [inline guides/coveralls-ci-cd/coveralls_ci_cd.md]: Add that they should add the coverage folder to gitignore

File: guides/coveralls-ci-cd/coveralls_ci_cd.md [hunk 4]
```diff
@@ -76,19 +77,57 @@ or if you're using Rails 5+ you can do
...
+
+![](https://raw.githubusercontent.com/CraftAcademyLabs/ca_course/c9bc41ab6e545fa807f956880f63ddd9a300c336/guides/coveralls-ci-cd/17_coveralls-enabled.png)
+
+At this point, we have successfully added coveralls to our project. If you need to have the coveralls token for continuous integration (Semaphore), click on the Details button.
+
+If you recently added coveralls to the repository you will be presented by the coveralls `repo_token` straight away. 
+
+![](https://raw.githubusercontent.com/CraftAcademyLabs/ca_course/c9bc41ab6e545fa807f956880f63ddd9a300c336/guides/coveralls-ci-cd/18_coveralls-repo-token.png)
+
+If you don't see it when you press details. Then you have to go to the settings for this project to find the coveralls `repo_token`.
```
Comment (GergKllai1) [inline guides/coveralls-ci-cd/coveralls_ci_cd.md]: Don't split this into 2 senteces

File: guides/coveralls-ci-cd/coveralls_ci_cd.md (hunk 1 above)
Comment (oliverochman) [reply guides/coveralls-ci-cd/coveralls_ci_cd.md]: fixed

File: guides/coveralls-ci-cd/coveralls_ci_cd.md (hunk 2 above)
Comment (oliverochman) [reply guides/coveralls-ci-cd/coveralls_ci_cd.md]: fixed

File: guides/coveralls-ci-cd/coveralls_ci_cd.md (hunk 3 above)
Comment (oliverochman) [reply guides/coveralls-ci-cd/coveralls_ci_cd.md]: fixed

File: guides/coveralls-ci-cd/coveralls_ci_cd.md (hunk 4 above)
Comment (oliverochman) [reply guides/coveralls-ci-cd/coveralls_ci_cd.md]: fixed

**Pull request merged**